src/http_rest_api_handler.cpp

```diff
@@ -131,9 +138,104 @@ void HttpRestApiHandler::registerAll() {
+            if (i->value.IsBool()) {
+                Value value(i->value.GetBool());
+                param.AddMember("bool_param", value, doc.GetAllocator());
+            }
+            scope["parameters"].GetObject()[i->name.GetString()] = param;
+        }
+    }
+}
+
+std::string HttpRestApiHandler::preprocessInferRequest(std::string request_body) {
+    static std::map<std::string, std::string> types = {
+        {"BOOL", "bool_contents"},
+        {"INT8", "int_contents"},
+        {"INT16", "int_contents"},
+        {"INT32", "int_contents"},
+        {"INT64", "int64_contents"},
+        {"UINT8", "uint_contents"},
+        {"UINT16", "uint_contents"},
+        {"UINT32", "uint_contents"},
+        {"UINT64", "uint64_contents"},
+        {"FP32", "fp32_contents"},
+        {"FP64", "fp64_contents"},
+        {"BYTES", "bytes_contents"}};
+
+    Document doc;
+    doc.Parse(request_body.c_str());
+    Value& inputs = doc["inputs"];
+    for (SizeType i = 0; i < inputs.Size(); i++) {
+        Value data = inputs[i].GetObject()["data"].GetArray();
+        Value contents(rapidjson::kObjectType);
+        Value datatype(types[inputs[i].GetObject()["datatype"].GetString()].c_str(), doc.GetAllocator());
+        contents.AddMember(datatype, data, doc.GetAllocator());
+        inputs[i].AddMember("contents", contents, doc.GetAllocator());
+        parseParams(inputs[i], doc);
+    }
+    Value& outputs = doc["outputs"];
+    for (SizeType i = 0; i < outputs.Size(); i++) {
+        parseParams(outputs[i], doc);
+    }
+    parseParams(doc, doc);
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    doc.Accept(writer);
+
+    return buffer.GetString();
+}
+
+::inference::ModelInferRequest HttpRestApiHandler::prepareGrpcRequest(const std::string modelName, const std::string modelVersion, const std::string request_body) {
+    ::inference::ModelInferRequest grpc_request;
+
+    std::string request(preprocessInferRequest(request_body));
+
+    google::protobuf::util::JsonParseOptions opts;
+    opts.ignore_unknown_fields = true;
+    google::protobuf::util::JsonStringToMessage(request, &grpc_request, opts);
+    grpc_request.set_model_name(modelName);
+    grpc_request.set_model_version(modelVersion);
+    return grpc_request;
+}
+
+Status HttpRestApiHandler::processInferKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& request_body) {
+    std::string modelName(request_components.model_name);
+    std::string modelVersion(std::to_string(request_components.model_version.value_or(0)));
+    SPDLOG_DEBUG("Processing REST request for model: {}; version: {}", modelName, modelVersion);
+    ::inference::ModelInferRequest grpc_request(prepareGrpcRequest(modelName, modelVersion, request_body));
+    ::inference::ModelInferResponse grpc_response;
+    const ::grpc::Status gstatus = kfsGrpcImpl.ModelInfer(nullptr, &grpc_request, &grpc_response);
```

Comment:
We cannot always respond with the same status code when inference does not pass. Probably we need to:
1) change grpc prediction service to have function that returns ovms::Status, and second one calling it transforming ovms::Status -> grpc::Status.

This comment applies as well to ModelReady.

Reply:
This can be fixed in further MR.